plugins/default/lib/mongodb_health.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+import argparse
+import sys
+import traceback
+from collections import Counter
+
+import openshift
+import nagios
+from mongodb import ReplsetStatuses
+
+
+def generate_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-s', '--selector', type=str, action="store",
+                        required=True, help="Label Selector")
+    return parser
+
+
+check_mongodb_cmd = ("bash", "-c",
+                     "mongo 127.0.0.1/admin "
+                     "-u \"admin\" -p \"$MONGODB_ADMIN_PASSWORD\" "
+                     "--eval=\"rs.status().myState\" --quiet")
+
+
+def parse_mongo_result(output):
+    try:
+        return int(output.rstrip("\n"))
+    except:
+        return None
+
+
+def analize(status):
+    if status is None:
+        return nagios.UNKNOWN
+    elif status == ReplsetStatuses.PRIMARY or status == ReplsetStatuses.SECONDARY:
+        return nagios.OK
+    elif status == ReplsetStatuses.ARBITER or status == ReplsetStatuses.STARTUP:
+        return nagios.WARN
+    else:
```

Comment:
- No need for `else` nor the `el` of `elif`, for we're returning in the if blocks.
- It is strange that we say "for anything else return CRITICAL" -- I'd say the logical choice here is "UNKNOWN"? IOW, check explicitly for what makes it OK, WARNING and CRITICAL, everything else is UNKNOWN.


```python
if status in (PRIMARY, SECONDARY):
    return nagios.OK
if status in (ARBITER, STARTUP):
    return nagios.WARN
if ...:
    return nagios.CRIT
return nagios.UNKNOWN
```

Or using a dict we get rid of all ifs:

```python
def analize(status):
    return {
        PRIMARY: nagios.OK,
        SECONDARY: nagios.OK,
        ARBITER: nagios.WARN,
        STARTUP: nagios.WARN,
        ...: nagios.CRIT,
    }.get(status, nagios.UNKNOWN)
```